edit/cms_functions.php

```diff
@@ -0,0 +1,478 @@
+    $output .= '<input type="submit" value="Edit"></form>';
+    return $output;
+}
+
+/*
+ * Displays the form for editing contact information. If there is a valid entry in the database, the values of each input field are filled out
+ * with the appropriate data.
+ *
+ * @param int $id The given entry to edit. This value is 0 if we are adding a new entry.
+ *
+ * @return Returns the input element for the given table, formatted as HTML. This should be placed inside the form element, before the submit
+ * element.
+ */
+function displayEditContactInfo(int $id) {
+    $array = getContactInfoFromDB($id);
+    $output = '<div class="longinput"><label>Icon ID: </label><input name="icon_id" type="text" value="' . $array['icon_id'] . '"></div>';
+    $output .= '<div class="longinput"><label>Link: </label><input name="link" type="text" value="' . $array['link'] . '"></div>';
+    $output .= '<div class="longinput"><label>Text: </label><input name="text" type="text" value="' . $array['text'] . '"></div>';
+    if($id != -1) {
+        $output .= '<input type="hidden" name="id" value ="' . $id . '">';
+    }
+    return $output;
+}
+
+function processMessage(int $id) {
+    switch ($id){
+        case 1: return 'About Me updated successfully!';
+        case 2: return 'Badge added successfully!';
+        case 3: return 'Badge edited successfully!';
+        case 4: return 'Project added successfully!';
+        case 5: return 'Project edited successfully!';
+        case 6: return 'Contact Info added successfully!';
+        case 7: return 'Contact Info edited successfully!';
+        case 8: return 'Icon added successfully!';
+        case 9: return 'Icon edited successfully!';
+        case 10: return 'Required field missing. Please try again.';
+        case 11: return 'Field not in valid format. Please try again.';
+        case 12: return 'Error adding entry into database. Please try again.';
+        case 13: return 'Error updating entry in database. Please try again.';
+        case 14: return 'Error deleting entry from database.';
+        case 15: return 'Deleted entry from the database';
+        case 16: return 'Please fill in all fields.';
+    }
+    return '';
+}
+
+/*
+ * Produces the list of entries on each Edit page, with accompanying edit/delete buttons
+ *
+ * @param string $table The table to get data from.
+ *
+ * @return Returns the list of entries for the given table, formatted as HTML. If we try a table where there is only a single row,
+ * such as the about table, we will receive an informative message instead telling us that this operation is not valid.
+ */
+function displayListHolderData(string $table, int $highlight = -1) {
+    if(!validateListTableRequest($table)) {
+        return "<div>This section cannot be listed.</div>";
+    }
+    $output = '<ul><li>';
+    //i am aware using * is a cardinal sin but I see no better way of doing what I want to do
+    $stmt = getPDO()->prepare('SELECT * FROM ' . $table);
```

Comment:
create a multi dimensional array with the key names like:
```
$tables = [
'table1' => ['field1', 'field2']
]
```
then you can use the table name to look up the fields you need and create your query from that

Reply:
resolved in f9e691a